ldm/modules/textual_inversion_manager.py

```diff
@@ -112,19 +113,21 @@ def load_textual_inversion(self, ckpt_path: Union[str,Path], defer_injecting_tok
             )
 
     def _add_textual_inversion(
-        self, trigger_str, embedding, defer_injecting_tokens=False
+            self, trigger_str, embedding, defer_injecting_tokens=False, source_file=Path
```

Comment:
i'm unhappy with leaking the `source_file` into this lower level function. IMO calling `_add_textual_inversion` with an already-in-use trigger_str *should* be an error, and the onus should be on the caller of `_add_textual_inversion` to fix it

Reply:
by which i mean - please remove the `source_file` arg and instead the caller of this function should ensure it is not trying to overwrite an already-loaded textual inversion

Reply:
I have moved the trigger munging logic into the `load_textual_inversion()` method and have removed the `source_file` arg as per recommendation.

I also have attached a dict to the TextualInversionManager object which keeps a mapping between trigger terms and the source file that requested them. This is for the sole purpose of status reporting. Here's how it looks with HuggingFace style `learned_embeds.bin` files:

```
>> redundant-style/learned_embeds.bin: Trigger token '<kamon-style>' is already claimed by 'kamon-style/learned_embeds.bin'. Trigger this concept with <redundant-style>
>> duplicate-style/learned_embeds.bin: Trigger token '<kamon-style>' is already claimed by 'kamon-style/learned_embeds.bin'. Trigger this concept with <duplicate-style>
>> Textual inversion triggers: <duplicate-style>, <kamon-style>, <redundant-style>
```

With .pt files, which usually have unique names, only the filename appears in the error messages. If there is still a conflict with the trigger term, which might happen if two .pt files with identical names appear in different directories during the recursive scan, then the system issues an error message and refuses to overwrite the first embedding, as per the original logic.

Also notice that